### PR TITLE
frontend: validate devicename input field in setup

### PR DESCRIPTION
**Pull request by thisconnect**
We already have a backend validation, but that only shows an invalid input message after the user clicked continue.

Better experience is to give live feedback and information about what is wrong.

It should only allow: printable ascii chars, with max length of 30 including spaces, but not starting or ending with (that is just trimmed, without showing an error to the user).

### CODE REVIEW

File: frontends/web/src/locales/en/app.json [hunk 1]
```diff
@@ -265,6 +265,11 @@
     },
     "stepCreate": {
       "description": "This name is used as the device name and for the backup.",
+      "error": {
+        "genericMessage": "For maximum compatibility only printable ascii characters and spaces are supported, max 30 chars.",
```
Comment (benma) [inline frontends/web/src/locales/en/app.json:269]: why 'maximum compatibility'?

ascii is ASCII uppercase.

File: frontends/web/src/locales/en/app.json [hunk 2]
```diff
@@ -265,6 +265,11 @@
...
+        "invalid-chars": "Name contains invalid characters.",
```
Comment (benma) [inline frontends/web/src/locales/en/app.json:270]: invalidChars

File: frontends/web/src/locales/en/app.json [hunk 3]
```diff
@@ -265,6 +265,11 @@
...
+        "too-long": "Name is too long."
```
Comment (benma) [inline frontends/web/src/locales/en/app.json:271]: tooLong

**Review by benma — APPROVED** [review]

File: frontends/web/src/routes/device/bitbox02/setup/name.tsx [hunk 4]
```diff
@@ -65,20 +80,25 @@ export const SetDeviceName = ({
...
           )}
         </ViewHeader>
-        <ViewContent minHeight="90px">
+        <ViewContent textAlign="left" minHeight="140px">
           <Input
             autoFocus
-            className={style.wizardLabel}
+            className={`${style.wizardLabel} ${error ? style.inputError : ''}`}
             label={t('bitbox02Wizard.stepCreate.nameLabel')}
-            pattern="^.{0,63}$"
```
Comment (benma) [inline frontends/web/src/routes/device/bitbox02/setup/name.tsx:73]: why not `pattern={regexPrintableAsciiOrSpace}`?

File: frontends/web/src/routes/device/bitbox02/setup/name.tsx [hunk 5]
```diff
@@ -38,10 +41,22 @@ export const SetDeviceName = ({
...
+    const valueTrimmed: string = value.trim();
+
+    if (valueTrimmed.length < 1) {
+      setError(undefined);
+    } else if (valueTrimmed.length > 30) {
+      setError('too-long');
+    } else if (!regexPrintableAsciiOrSpace.test(valueTrimmed)) {
+      setError('invalid-chars');
+    } else {
+      setError(false);
```
Comment (benma) [inline frontends/web/src/routes/device/bitbox02/setup/name.tsx:58]: `return` after setting the error? calling `setDeviceName(...)` will fail anyway.

Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox02/setup/name.tsx:58]: well need deviceName always to be the current value. As it is later always passed back to the Input field no matter if it is valid or not.

https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable

Comment (benma) [reply frontends/web/src/routes/device/bitbox02/setup/name.tsx:58]: ah i misread and thought this invokes the set device name thing. nvm :+1:

**Comment by thisconnect** [comment]
Screen Shot 2024-01-25 at 16 35 40

Screen Shot 2024-01-25 at 16 35 54

Screen Shot 2024-01-25 at 16 36 15

Screen Shot 2024-01-25 at 16 36 26

File: frontends/web/src/locales/en/app.json [hunk 6]
```diff
@@ -265,6 +265,11 @@
     },
     "stepCreate": {
       "description": "This name is used as the device name and for the backup.",
+      "error": {
+        "genericMessage": "For maximum compatibility only ASCII characters and spaces are supported, max 30 chars.",
```
Comment (benma) [inline frontends/web/src/locales/en/app.json:269]: same as before, please remove 'maximum compatibility', it kind of does not mean anything and just adds work to translators. 

i was wondering if ASCII might not be understandable to a normal uset. chatgpt suggets

> Use letters, numbers, basic symbols, spaces. Max 30 characters.

**Review by benma — APPROVED** [review]
very nice! please squash in the end

**Pull request merged**